### PR TITLE
[IMP] e-commerce : click and drag implemented for top categories tab

### PR DESCRIPTION
**Pull request by luve-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale/static/src/scss/website_sale.scss [hunk 1]
```diff
@@ -86,11 +86,16 @@ $o-wsale-products-layout-grid-gutter-width: min($grid-gutter-width / 2, $o-wsale
             transform: translateX(0);
 
             li {
-                scroll-snap-align: start;
+                scroll-snap-align: none;
```
Comment (Williambraecky) [inline addons/website_sale/static/src/scss/website_sale.scss:89]: We can remove the whole rule

File: addons/website_sale/static/src/js/website_sale.js [hunk 2]
```diff
@@ -301,6 +310,47 @@ publicWidget.registry.WebsiteSale = publicWidget.Widget.extend(VariantMixin, car
...
+        ev.currentTarget.scrollLeft = this.scrollLeft - walk;
+    },
+
+    _onClickHandler: function(ev) {
+        if(this.moved) {
+            ev.stopPropagation();
+            ev.preventDefault();
+        }
+    },
+
```
Comment (Williambraecky) [inline addons/website_sale/static/src/js/website_sale.js:353]: Linting
```suggestion
    _onMouseDown: function (ev) {
        this.down = true;
        this.drag = false;
        this.startX = ev.pageX - ev.currentTarget.offsetLeft;
        this.scrollLeft = ev.currentTarget.scrollLeft;
        this.formerTarget = ev.target;
        this.moved = false;
    },

    _onMouseLeave: function (ev) {
        if (!this.down) {
            return;
        }
        ev.currentTarget.classList.remove('activeDrag');
        this.down = false
    },

    _onMouseUp: function (ev) {
        this.down = false;
        ev.currentTarget.classList.remove('activeDrag');
    },

    _onMouseMove: function (ev) {
        if (!this.down) {
            return;
        }
        ev.preventDefault();
        ev.currentTarget.classList.add('activeDrag');
        this.moved = true;
        const x = ev.pageX - ev.currentTarget.offsetLeft;
        const walk = (x - this.startX) * 2;
        ev.currentTarget.scrollLeft = this.scrollLeft - walk;
    },

    _onClickHandler: function(ev) {
        if(this.moved) {
            ev.stopPropagation();
            ev.preventDefault();
        }
    },

```

File: addons/website_sale/static/src/js/website_sale.js [hunk 3]
```diff
@@ -241,6 +246,10 @@ publicWidget.registry.WebsiteSale = publicWidget.Widget.extend(VariantMixin, car
         this._changeCountry = _.debounce(this._changeCountry.bind(this), 500);
 
         this.isWebsite = true;
+        this.startX = 0;
+        this.isDown = false;
+        this.scrollLeft = 0;
+        this.moved = false;
```
Comment (Williambraecky) [inline addons/website_sale/static/src/js/website_sale.js:252]: Make it explicit these are related to the filmstrip drag mechanic
```suggestion
        // Filmstrip drag&drop
        this.filmStripStartX = 0;
        this.filmStripIsDown = false;
        this.filmStripScrollLeft = 0;
        this.filmStripMoved = false;
```
I agree those are long variable names but we are already working with a gigantic file handling a lot of different pages. It makes it way easier to identify and separate part of the code.

**Review by Williambraecky — COMMENTED** [review]
Thanks for your work :)

**Comment by luve-odoo** [comment]
done ! :)

**Review by Williambraecky — APPROVED** [review]
Thanks for your work
robodoo r+

**Pull request closed without merging**